pkg/scheduler/algorithm/priorities/selector_spreading.go

```diff
@@ -211,7 +211,9 @@ func filteredPod(namespace string, selector labels.Selector, nodeInfo *scheduler
 		return []*v1.Pod{}
 	}
 	for _, pod := range nodeInfo.Pods() {
-		if namespace == pod.Namespace && selector.Matches(labels.Set(pod.Labels)) {
+		// Ignore pods being deleted for spreading purposes
+		// Similar to how it is done for SelectorSpreadPriority
+		if namespace == pod.Namespace && pod.DeletionTimestamp == nil && selector.Matches(labels.Set(pod.Labels)) {
```

Comment:
Does this affect the default scheduling based on region and zone labels (failure-domain.beta.kubernetes.io/region)  ?
Does this mean that without this fix , if a pod is currently in terminating condition , it would count as a pod on the node and hence additional pods may not be placed on that node ? @aveshagarwal @bsalamat